airbyte-integrations/connectors/source-postgres/src/main/java/io/airbyte/integrations/source/postgres/PostgresSource.java

```diff
@@ -404,6 +408,25 @@ private static AirbyteStream addCdcMetadataColumns(final AirbyteStream stream) {
     return stream;
   }
 
+  // TODO This is a temporary override so that the Postgres source can take advantage of per-stream
+  // state.
+  @Override
+  protected List<AirbyteStateMessage> deserializeState(final JsonNode stateJson) {
+    if (stateJson == null) {
+      // TODO What should the default/empty state be -- per stream or global?
```

Comment:
We should check if CDC is active or not. If active, choose global else choose per stream

Reply:
@subodh1810 I realized this last night.  I have already pushed up a change to check and return the global state manager if CDC is enabled.